### ISSUE TITLE
Installation Issues

### ISSUE THREAD

**Issue by tiaroun** [OpenNFT/OpenNFT#161]
Hello!

I originally encountered the same issue with Importing DistUtilsOptionError described in a previous Issue (Issue #160) while trying to install OpenNFT on a Windows 11 laptop whilst running Python 3.7.9. I tried to troubleshoot with the suggestion referenced below: 

              It is a version issue, and I have also encountered it. The quick fix is to edit https://github.com/OpenNFT/OpenNFT/blob/30e784adf92c234eeb97efc86c6d72cf449aa9fc/setup.py#L10 by changing `distutils.core` to `distutils.errors`.

_Originally posted by @tiborauer in https://github.com/OpenNFT/OpenNFT/issues/160#issuecomment-1551835699_
            
however, I have now encountered a new error: 
```

(opennft-venv) C:\> pip install C:\OpenNFT\
Processing c:\opennft
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [13 lines of output]
      c:\opennft-venv\lib\site-packages\setuptools\__init__.py:84: _DeprecatedInstaller: setuptools.installer and fetch_build_eggs are deprecated.
      !!

              ********************************************************************************
              Requirements should be satisfied by a PEP 517 installer.
              If you are using pip, you can try `pip install --use-pep517`.
              ********************************************************************************

      !!
        dist.fetch_build_eggs(dist.setup_requires)
      error in OpenNFT setup command: 'install_requires' must be a string or list of strings containing valid project/version requirement specifiers; Local version label can only be used with `==` or `!=` operators
          numpy>=1.13.1,>=1.13.1+mkl
                        ~~~~~~~~^
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```
Running the pip install --use-pep517 tip in the error message just returned another error. I am not sure how to resolve this. Any help would be appreciated!

**Comment by ykoush** [comment]
Hi, sounds as compatibility issues that need to be resolved, perhaps @espdev or @tiborauer could comment and @espdev could help adjust the installer accordingly.
Thanks.

**Comment by espdev** [comment]
It seems the version label `1.13.1+mkl` is not correct for new pip. 
I think we can try to use `intel-numpy` wheel on Windows instead: https://pypi.org/project/intel-numpy

**Comment by ykoush** [comment]
@espdev please explain what do you mean, like a sequence of steps to try, or fix?

**Comment by tiborauer** [comment]
Replace "numpy>=1.13.1,>=1.13.1+mkl" with "intel-numpy>=1.31.1" (I assume we should use the same versioning) in requirements.txt.

**Comment by Kanzuli** [comment]
Hey, I ran into issues when I tried to replace "numpy>=1.13.1,>=1.13.1+mkl" in the requirements.txt with "intel-numpy>=1.13.1". Everything was installed nicely but I could not run software with that numpy for some reason. After digging up a while, I found a compatible version (with my python 3.8) of numpy+mkl here: https://www.lfd.uci.edu/~gohlke/pythonlibs/#numpy which I then manually installed after the commands:
```
pip install -U pip setuptools wheel
pip install -e .
```
First I downloaded a correct file and moved it under OpenNFT folder and installed in the virtual environment:

`pip install -U numpy-1.22.4+mkl-cp38-cp38-win_amd64.whl`

Note, here cp38 means python version 3.8 etc.

**Comment by manish-alluri** [comment]
Hello team,

I have successfully installed and have used OpenNFT previously in 2023 for our real-time project and now have a new project for which we want to use this software again. I have tried installing it multiple times and with the fixes on this thread working for a clean install of OpenNFT but when it came to the installation of the **matlabengineforpython** which I was looking for to run opennft, I was not able to find it even after having a clean install. 

We are currently using the **Matlab R2023b** version which I see is only compatible starting from python 3.9 so I have tried creating and installing it on both 3.8 and 3.9 envs but still did not install the Matlab engine. Is this a versioning issue (which I'm assuming) or Is there a way to fix this to work for Matlab 2023b as well? I hope I could get some help relating to this issue which is something new.

Thanks.
